Jd/psy45 docs (#1385)
NREL-Sienna/PowerSystems.jl: docs/src/explanation/system.md

 1. We make no guarantees on the stability of component structure definitions. We will maintain version stability on the accessor methods.
 2. Per-unit conversions are made in the return of data from the accessor functions. (see the [per-unit section](@ref per_unit) for more details)

> ## [Using sub-systems](@id subsystems)

Author: @daniel-thom

> ## [Using sub-systems](@id subsystems)

Reviewer: This section would probably better fit as a tutorial, instead of being in Explanation ("Creating and Using Subsystems")? I'd split up the code chunks more, with prompts to make it very obvious what's going on: "Let's retrieve the generator named Alta and add it to subsystem 1:"

Reviewer: Also, `add_system` and [from_subsystem](https://nrel-sienna.github.io/PowerSystems.jl/previews/PR1385/api/public/#PowerSystems.from_subsystem-Tuple{System,%20AbstractString}) have bare-bones docstrings, and could use some of the information that's here

in aggregations that are not related to the existing Areas or LoadZones mappings. The subsystems are assigned using any arbitrary string.
It is possible to add subsystems by calling the following commands on the system:

```@repl sub-system
using PowerSystems;
using PowerSystemCaseBuilder;
sys = build_system(PSISystems, "c_sys5_pjm")
add_subsystem!(sys, "1")
add_subsystem!(sys, "2")
```

Devices in the system can be assigned to the subsystems in the following way using the function [`add_component_to_subsystem!`](@ref)

```@repl subsystem
g = get_component(ThermalStandard, sys, "Alta")
add_component_to_subsystem!(sys, "1", g)

g = get_component(ThermalStandard, sys, "Sundance")
add_component_to_subsystem!(sys, "2", g)
```

To retrieve components assigned to specific subsystem the use the keyword argument `subsystem_name`.

```@repl subsystem
gens_1 = get_components(ThermalStandard, sys; subsystem_name = "1")
get_name.(gens_1)

gens_2 = get_components(ThermalStandard, sys; subsystem_name = "2")
get_name.(gens_2)
```

One useful feature that requires care when used is generating a new system from a subsystem assignment.

> The function [`from_subsystem`](@ref) will allow the user to produce a new system object that can be used or exported.

Reviewer: ```suggestion
The function [`from_subsystem`](@ref) will allow the user to produce a new [`System`](@ref) that can be used or exported.
```

> This functionality requires careful subsystem assignemnt of the devices and its dependencies. In the current example this code will as follows

Reviewer 2: Are we ever using second person point of view? If not, my suggestion needs adjustment.

> This functionality requires careful subsystem assignemnt of the devices and its dependencies. In the current example this code will as follows

Reviewer: system -> [`System`](@ref)

```@repl subsystem
from_subsystem(sys, "1")
```

> In order to fix the error the bus in the Alta generator also needs to be added to the subsystem 1.

Reviewer 2: The system is invalid because the bus connected to the Alta generator is not part of the subsystem. We can add it, and then run `from_subsystem` again.

```@repl subsystem
g = get_component(ThermalStandard, sys, "Alta")
b = get_bus(g)
add_component_to_subsystem!(sys, "1", b)
from_subsystem(sys, "1")
```

> Advanced users can use the keyword `runchecks=false` and avoid any topological check in the process.

Reviewer: Move these to the docstring and incorporate into an arguments list

> It is highly recommended that users do this only if there is a clear understanding the resulting system might be invalid for any modeling.

Reviewer 2: ```suggestion
It is highly recommended that users only do this if they clearly understand how to validate the resulting system before using it for modeling.
```

Reviewer 2: What is our standard for using admonitions?

Have we given sufficient guidance for validation?
